src/app/my-info/(account)/wishlist/_components/WishList.tsx

```diff
@@ -0,0 +1,160 @@
+'use client';
+
+import { deleteProductLikes, getProductLikes } from '@/api/likesAPI';
+import { MyInfoEmptyCase } from '@/app/my-info/_components';
+import { Button, Dialog } from '@/components';
+import { QUERY_KEYS } from '@/constants/queryKey';
+import { GetProductLikesParams, WishlistPageProps } from '@/types/LikeTypes';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import classNames from 'classnames/bind';
+import { ChangeEvent, useState } from 'react';
+import { toast } from 'react-toastify';
+import WishItem from './WishItem';
+import styles from './WishList.module.scss';
+
+const cn = classNames.bind(styles);
+
+export default function WishList({ searchParams }: WishlistPageProps) {
+  const queryClient = useQueryClient();
+  const [selectedList, setSelectedList] = useState<Set<number>>(new Set());
+  const [isDeleteSelectedOpen, setIsDeleteSelectedOpen] = useState(false);
+  const [isDeleteAllOpen, setIsDeleteAllOpen] = useState(false);
+
+  const getProductLikesParams: GetProductLikesParams = {
+    page: searchParams.page || '0',
+    size: searchParams.size || '10',
+  };
+
+  const { data, isLoading } = useQuery({
+    queryKey: [...QUERY_KEYS.LIKE.LISTS(), searchParams],
+    queryFn: () => getProductLikes(getProductLikesParams),
+  });
+
+  const { mutate: deleteWishItem } = useMutation({
+    mutationFn: deleteProductLikes,
+    onSuccess: () => {
+      toast.success('삭제를 완료 했습니다.');
+      queryClient.invalidateQueries({
+        queryKey: QUERY_KEYS.LIKE.LISTS(),
+      });
+
+      setIsDeleteSelectedOpen(false);
+      setIsDeleteAllOpen(false);
+      setSelectedList(new Set());
+    },
+  });
+
+  if (isLoading) {
+    return null;
+  }
+
+  if (data?.length === 0) {
+    return <MyInfoEmptyCase message='찜한 상품이 없습니다.' />;
+  }
+
+  const updateSet = (set: Set<number>, id: number) => {
+    const updatedSet = new Set(set);
+
+    if (updatedSet.has(id)) {
+      updatedSet.delete(id);
+    } else {
+      updatedSet.add(id);
+    }
+
+    return updatedSet;
+  };
+
+  const handleOnChange = (id: number) => {
+    setSelectedList((prevSet) => updateSet(prevSet, id));
+  };
+  const toggleAllCheckedById = (e: ChangeEvent<HTMLInputElement>) => {
+    const { checked } = e.target;
+    if (checked) {
+      const allChecked = new Set(data?.map(({ productId }) => productId));
+      setSelectedList(allChecked);
+    } else {
+      setSelectedList(new Set());
+    }
+  };
+
+  const isAllSelected = () => data?.every((v) => selectedList.has(v.productId));
+
+  const handleAllDelete = () => {
+    if (!data) {
+      return;
+    }
+    deleteWishItem(data.map(({ productId }) => productId));
```

Comment:
```suggestion
  const handleAllDelete = () => {
    if (data) {
       deleteWishItem(data.map(({ productId }) => productId));
    
```
이렇게 해도 괜찮을 것 같다는 개인적인 생각입니다~